### DISCUSSION TITLE
Welcome to hoermann_door Discussions!

### DISCUSSION THREAD

**Discussion by stephan192** [stephan192/hoermann_door#20] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by pabutterworth** [comment]
Hi, Do you know if anyone has any of these PCB's built and for sale? Would love one and am not competent enough to manufacture myself!
Thanks
Paul

**Reply by stephan192** [reply]
From time to time people offer spare boards on the [home assistant forum](https://community.home-assistant.io/t/hormann-garage-door-via-mqtt/279938). But you must be carefull which door you have. My project is for version 3 drives, but half of the posts in the forum concern version 4 drives.

**Reply by pabutterworth** [reply]
Mine is a version 3 also. If it was a version 4, I would take on the build, but I cant solder SMD stuff!

**Reply by Gamogamer** [reply]
I made some boards for version three but then found out that I actually have version two and I couldn't get it to work so I have 4 boards around

**Comment by pabutterworth** [comment]
Ah are they complete, how much do you want for one?
Thanks
Paul

On Wed, 14 Jun 2023 at 23:45, Gamogamer ***@***.***> wrote:

> I made some boards for version three but then found out that I actually
> have version two and I couldn't get it to work so I have 4 boards around
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/stephan192/hoermann_door/discussions/20#discussioncomment-6179843>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD737V6PUISEKMRX76M2KFLXLI5IVANCNFSM6AAAAAAYOOW4PM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Paul Butterworth
+44 7768 252250

**Reply by Gamogamer** [reply]
While I have all the parts they are not fully assembled. It depends on if you want just the PCB and parts or if you'd like me to assemble it also. Unassembled parts cost is around 25€. Do you have a pic programmer?

**Comment by pabutterworth** [comment]
I would prefer assembled, I can do normal soldering fine, but not SMD. No,
I don't have a PIC programmer either.

Would you be willing to build a complete programmed unit? if so what would
be the cost?

Thanks
Paul

On Fri, 16 Jun 2023 at 09:53, Gamogamer ***@***.***> wrote:

> While I have all the parts they are not fully assembled. It depends on if
> you want just the PCB and parts or if you'd like me to assemble it also.
> Unassembled parts cost is around 25€. Do you have a pic programmer?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/stephan192/hoermann_door/discussions/20#discussioncomment-6194736>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD737V7PLXUYCE62OSLIOQLXLQNIVANCNFSM6AAAAAAYOOW4PM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Paul Butterworth
+44 7768 252250

**Reply by Gamogamer** [reply]
Yea I can assemble one of the units for you and program the pic. You will need to program the ESP with your Credentials. I would do it for 45€+Shipping.

**Reply by surfmk** [reply]
Hi, I would be interested in an assembled hörmann supramatic e3 control ESP32 board that Stephan designed... do you have any left? Thanks, Paul

**Reply by Gamogamer** [reply]
Hi Surf, I still have some of the ones i made back then. But they are esp8266. I am not sure which boards are esp32.

**Reply by surfmk** [reply]
Hi, apologies I meant esp8266. Actually I'm more familiar with ESPHome / HASS and wondering if you would be able to flash the firmware, so it would be plug and play? I have only used YAML to build firmware and not the Arduino that Stephen references. I have the E3 Hoermann garage door unit.

**Reply by surfmk** [reply]
> Hi Surf, I still have some of the ones i made back then. But they are esp8266. I am not sure which boards are esp32.

Hi Gamogamer, I have studied Stephan's Wiki again and I can manage more than I thought. I have a USB-UART converter I used for flashing Sonoff POWA / NSPanel devices and I could install Arduino IDE 2.0 software and compile the new firmware (particular attention to credentials file). I can see SV3 is the TX/RX/Ground but where he says "Check if 5V and 3.3V are fine"... I am not sure exactly which pins are supplying these voltages (possibly supply to esp8266). So overall, I'm OK with the esp8266. 

My only issue would be Pic programming as I sold my PICkit3 in about 2005! I could potentially buy another one but I would be really grateful for programming the PIC for me as this is a one time operation.

Many thanks in advance,

Paul

**Reply by Gamogamer** [reply]
Yea I think I should be able to program the pic still, I'll just figure it out again. My board is slightly different than the original since the powerdelivery parts where unavailable back then. I'll set myself a reminder to look into it tomorrow.

**Reply by surfmk** [reply]
Thanks for getting back. OK, if you still have the parts that would be great. I live in UK, Seaford, East Sussex BN25 1BL for posting.
Just checking the wiki again. I will need : 6 pin to 6 pin RJ11 like this: [https://www.ebay.co.uk/itm/397219777285](url)
Happy to buy from eBay or add to your parts?

**Reply by Gamogamer** [reply]
I could not actually find those cables for sale back then so i jankily made my own, probably best you just get those. 
I checked my Project boxes and i still have 4 boards and parts for them. I just assembled one for you and will look into flashing the pic tomorrow. For flashing the ESP you can actually hot rod a nodemcu, thats what I have been using for years.

**Reply by surfmk** [reply]
Hi, thanks very much. I've been discussing how to connect up my USB-UART converter with Stephan and now have a clear picture what I'm doing. Nodemcu is new to me but I think I can see how you might do that... clever chap! Really appreciate your help here and look forward to getting it all together.

**Reply by Gamogamer** [reply]
Hi. I'm a bit later than expected but I've had surgery and didn't get the programmer to work when I tried last week, I'll try again in few days

**Reply by surfmk** [reply]
Hi, sorry to hear you've been under the knife... hope you're recovering well.
I have got another option if you are not sure about getting the PIC programmer working.
Apparently there is a guy selling ESP32 version that is compatible with E3 Hormann controller. He wants 45 Euros.
What were you thinking of charging?
Thanks,
Paul

**Reply by Gamogamer** [reply]
Hi Paul,

sorry it's been a while again. But I am Healthy again. Thank you for your wishes.
Turns out when you're sick you're no good at programming. I got it programmed and working now.
I understand if I'm too late now. Was gonna be the same including shipping probably.
Best, Robert

**Reply by surfmk** [reply]
Hi, thanks for reaching out again. I still have not bought a board. I am just looking at the best option as I have now learned a bit more about the ESP32 version which looks interesting. If you give me a few days, I hope to have a few questions answered.
I may still like to buy your board... so I will get back in touch soon.
